Airport Ticket Booking System/Services/PriceService.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace Airport_Ticket_Booking_System;
+
+public class PriceService : IPriceService
+{
+    private FlightPrice _flightPrice;
+
+    private const int _lateBooking = 30;
+    private const int _normalBooking = 60;
+    private const int _earlyBooking = 120;
+
+    private const decimal _feesForLateBooking = 1.2m;
+    private const decimal _feesForNormalBooking = 1.1m;
+    private const decimal _feesForWeekendBookings = 1.1m;
+    private const decimal _feesForHolidaysBooking = 1.3m;
+    private const decimal _discountForEarlyBooking = 0.8m;
+
+    public PriceService()
+    {
+    }
+
+    public decimal CalculateTotalPrice(Airlines airline, FlightClass flightClass, int numberOfAdults, int numberOfChildren,
+        int numberOfBabies, Currency targetCurrency, DateTime bookingDate, DateTime flightDate)
+    {
+
+        decimal baseTotalPrice = _flightPrice.GetTotalPrice(airline, flightClass, numberOfAdults, numberOfChildren, numberOfBabies, Currency.EUR);
+
+        decimal updatedPrice = AdditionalCharges(baseTotalPrice, bookingDate, flightDate);
+
+        return updatedPrice.ConvertToCurrency(Currency.EUR, targetCurrency);
+    }
+
+    public decimal AdditionalCharges(decimal basePrice, DateTime bookingDate, DateTime flightDate)
+    {
+        int daysUntilFlight = (flightDate - bookingDate).Days;
+
+        if (daysUntilFlight <= _lateBooking)
+            basePrice *= _feesForLateBooking;
+
+        else if (daysUntilFlight <= _normalBooking)
+            basePrice *= _feesForNormalBooking;
+
+        else if (daysUntilFlight >= _earlyBooking)
+            basePrice *= _discountForEarlyBooking;
+
+        if (IsHoliday(flightDate))
+            basePrice *= _feesForHolidaysBooking;
+
+        if (IsWeekend(flightDate.DayOfWeek))
+            basePrice *= _feesForWeekendBookings;
+
+        return basePrice;
+    }
+
+    private bool IsHoliday(DateTime date)
```

Comment:
This one and `IsWeekend` can be moved to a separate utility class, as they are generic methods that are not just specific to `PriceService`.